lib/identity_cache/belongs_to_caching.rb

```diff
@@ -34,14 +34,21 @@ def build_normalized_belongs_to_cache(association, options)
         foreign_key = options[:association_reflection].foreign_key
         self.class_eval(<<-CODE, __FILE__, __LINE__ + 1)
           def #{options[:cached_accessor_name]}
-            if IdentityCache.should_use_cache? && #{foreign_key}.present? && !association(:#{association}).loaded?
+            association_klass = association(:#{association}).klass
+            should_use_cache = if association_klass.respond_to?(:should_use_cache?)
```

Comment:
This shouldn't be necessary, since we already need the associated class to `include IdentityCache` to call `fetch_by_id` on it.